R/shiny_prerendered.R

```diff
@@ -0,0 +1,386 @@
+
+# Create a shiny app object from an Rmd w/ runtime: shiny_prerendered
+shiny_prerendered_app <- function(input_rmd, encoding, render_args) {
+
+  # get rendered html
+  html <- shiny_prerendered_html(input_rmd, encoding, render_args)
+
+  # create the server environment
+  server_envir = new.env(parent = globalenv())
+
+  # extract the server-start context
+  html_lines <- strsplit(html, "\\r?\\n")[[1]]
+  server_start_context <- shiny_prerendered_extract_context(html_lines,
+                                                            "server_start")
+  onStart <- function() {
+    shiny_prerendered_data_load(input_rmd, server_envir)
+    eval(parse(text = server_start_context), envir = server_envir)
+  }
+
+  # extract the server context
+  server_context <- shiny_prerendered_extract_context(html_lines, "server")
+  server_envir$server_context <- server_context
+  server <- function(input, output, session) {
+    eval(parse(text = server_context))
+  }
+  environment(server) <- server_envir
+
+  # create shiny app
+  shiny::shinyApp(
+    ui = html,
+    server = server,
+    onStart = onStart,
+    uiPattern = "^/$|^(/.*\\.[Rr][Mm][Dd])$"
+  )
+}
+
+
+# Generate the html for a runtime: shiny_prerendered Rmd (attempts to use
+# an existing rendering of the html if it's still valid)
+shiny_prerendered_html <- function(input_rmd, encoding, render_args) {
+
+  # determine the path to the rendered_html
+  output_file <- render_args$output_file
+  if (is.null(output_file))
+    output_file <- file_with_ext(basename(input_rmd), "html")
+  output_dir <- render_args$output_dir
+  if (is.null(output_dir))
+    output_dir <- dirname(input_rmd)
+  rendered_html <- file.path(output_dir, output_file)
+
+  # determine whether we need to render the Rmd in advance
+  prerender_option <- tolower(Sys.getenv("RMARKDOWN_RUN_PRERENDER", "1"))
```

Comment:
where does this get set? (or is it a hook for eventual use by the IDE and/or Connect?)